spec/grid_spec.rb

```diff
@@ -1,108 +1,132 @@
 require 'spec_helper'
-require_relative "../lib/grid"
+require_relative '../lib/grid'
+require_relative '../lib/ship'
 
 RSpec.describe Grid do
 
-  let(:grid) {Grid.new}
-  let(:game) {Game.new(grid)}
-  let(:player) {Player.new(grid)}
+  let(:grid) {Grid.new(10)}
+  let(:ship) {Ship.new("destroyer", 2)}
 
-  describe "registers positions of all ships placed" do
-    it "registers submarine with its position in the list of ships placed" do
-      submarine = Ship.new("submarine", 1)
+  it "has a size" do
+    expect(grid.size).to eq(10)
+  end
 
-      grid.mark_ship_positions(5, "A", submarine, :vertical)
+  describe "knows where ship can be placed" do
+    it "returns :invalid_ship_position if 1st coordinate is invalid number" do
+      ui_output = {:x => "a", :y => "B", :orientation => :horizontal}
+      result = grid.validate_placement(ui_output, ship.length)
 
-      expect(grid.ships_placed).to eq({submarine => [[5, "A"]]})
+      expect(result).to eq(:invalid_ship_position)
     end
 
-    it "registers destroyer with its position in the list of ships placed" do
-      destroyer = Ship.new("destroyer", 2)
-
-      grid.mark_ship_positions(5, "B", destroyer, :vertical)
+    it "returns false if 2nd coordinate is invalid letter" do
```

Comment:
Yes, I just didn't notice...same for below. I changed the method return type a lot of times!!